aten/src/ATen/native/cuda/ReflectionPadding1d.cu

```diff
@@ -0,0 +1,215 @@
+    nbatch = input_.size(0);
+    dim_plane++;
+    dim_w++;
+  }
+
+  int64_t pad_l = padding[0];
+  int64_t pad_r = padding[1];
+
+  int64_t nplane = input_.size(dim_plane);
+  int64_t input_w = input_.size(dim_w);
+  int64_t output_w  = input_w + pad_l + pad_r;
+
+  AT_CHECK(pad_l < input_w && pad_r < input_w, "Padding size should be less "
+    "than the corresponding input dimension, but got: padding (",  pad_l, ", ",
+    pad_r, ") at dimension ", dim_w, " of input ", input_);
+
+  AT_CHECK(output_w >= 1,
+    "input (W: ", input_w, ")is too small. Calculated output W: ", output_w);
+
+  if (input_.ndimension() == 2) {
+    output.resize_({nplane, output_w});
+  } else {
+    output.resize_({nbatch, nplane, output_w});
+  }
+
+  dim3 block_size(output_w > 256 ? 256 : output_w);
+  dim3 grid_size((int) ::ceil(output_w / 256.0), nplane, nbatch);
+
+  Tensor input = input_.contiguous();
+
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(
+    input.type(), "reflection_pad1d_out_template", [&] {
+      reflection_pad1d_out_kernel<<<
+        grid_size, block_size, 0, at::cuda::getCurrentCUDAStream()>>>(
+          input.data<scalar_t>(), output.data<scalar_t>(),
+          input_w, pad_l, pad_r);
+    }
+  );
+
+  AT_CUDA_CHECK(cudaGetLastError());
+}
+
+void reflection_pad1d_backward_out_template(
+    Tensor & grad_input, const Tensor & grad_output_,
+    const Tensor & input, IntList padding) {
+
+  AT_CHECK(canUse32BitIndexMath(input),
+    "input tensor must fit into 32-bit index math");
+
+  AT_CHECK(canUse32BitIndexMath(grad_output_),
+    "input tensor must fit into 32-bit index math");
+
+  int64_t dim_plane = 0;
+  int64_t dim_w = 1;
+  int64_t nbatch = 1;
+
+  if (input.ndimension() == 3) {
+    nbatch = input.size(0);
+    dim_plane++;
+    dim_w++;
+  }
+
+  int64_t pad_l = padding[0];
+  int64_t pad_r = padding[1];
+
+  int64_t nplane = input.size(dim_plane);
+  int64_t input_w = input.size(dim_w);
+  int64_t output_w  = input_w + pad_l + pad_r;
+
+  Tensor grad_output = grad_output_.contiguous();
+
+  AT_CHECK(output_w == grad_output.size(dim_w),
+    "gradOutput width unexpected. Expected: ", output_w, ", Got: ",
+    grad_output.size(dim_w));
+
+  dim3 block_size(output_w > 256 ? 256 : output_w);
+  dim3 grid_size((int) ::ceil(output_w / 256.0), nplane, nbatch);
+
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(
+    input.type(), "reflection_pad1d_backward_out_template", [&] {
+      reflection_pad1d_backward_out_kernel<<<
+        grid_size, block_size, 0, at::cuda::getCurrentCUDAStream()>>>(
+          grad_input.data<scalar_t>(), grad_output.data<scalar_t>(),
+          input_w, pad_l, pad_r);
+    }
+  );
+
+  AT_CUDA_CHECK(cudaGetLastError());
+}
+
+} // namespace
+
+Tensor& reflection_pad1d_out_cuda(
+    Tensor& output, const Tensor& input, IntList padding) {
+  reflection_pad1d_out_template(output, input, padding);
+  return output;
+}
+
+Tensor reflection_pad1d_cuda(Tensor const& input, IntList padding) {
```

Comment:
nit: can you be consistent with "Tensor const" vs "const Tensor" -- I believe NativeFunctions.h gets generated as "const Tensor&" so let's use that.